src/main/java/model/canvas/layer/ILayer.java

```diff
@@ -1,10 +1,44 @@
 package model.canvas.layer;
 
+import java.awt.Color;
+
 /**
- * The {@code ILayer} interface specifies objects that represent some sort of "layer" in the
+ * The {@code ILayer} interface specifies objects that represent some sort of layer. This interface
+ * is a subinterface to {@code IReadOnlyLayer}.
+ *
+ * @see IReadOnlyLayer
  */
-public interface ILayer {
+public interface ILayer extends IReadOnlyLayer {
+
+  /**
+   * Sets the pixel color at the specified coordinates.
+   *
+   * @param x the x-coordinate of the pixel that will be changed.
+   * @param y the y-coordinate of the pixel that will be changed.
+   * @param color the new color of the specified pixel.
+   * @throws IndexOutOfBoundsException if the specified coordinate is out-of-bounds.
+   * @throws NullPointerException if any arguments are {@code null}.
+   */
```

Comment:
Same as before: "Could be valuable to specify that the origin is at the top left corner and that the coordinates are zero-indexed."